km61/Kutsenko_Oleksandr/laboratory1.sql

```diff
@@ -1,2 +1,141 @@
 -- LABORATORY WORK 1
 -- BY Kutsenko_Oleksandr
+/*---------------------------------------------------------------------------
+1.Створити схему даних з назвою – прізвище студента, у якій користувач може: 
+створювати таблиці
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+CREATE USER kucenko
+IDENTIFIED BY kucenko
+DEFAULT TABLESPACE "kucenko"
+TEMPORARY TABLESPACE "kucenko"
+
+ALTER "CONNECT" TO kucenko
+QUOTA 100M TO kucenko;
+
+GRANT UPDATE ANY TABLE TO kucenko;
+
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+Студент має мобільний номер іноземного оператора.
+4 бали
+
+---------------------------------------------------------------------------*/
```

Comment:
There should be tables - "Students" and "Phones" with one to many connection, which represents that one student can have many different foreign phone numbers. So table "Phone"(being dependent from "Students") should have a field that is a reference on "Students" table(foreign key on "Students"). 

Students
-------------
(1) id - number(20) primary key not null, 
(2) first_name - varchar2(30) not null
Regexp - ^[A-Z]{1}[a-z]{29}+$ - defines first capital letter and 29 small letters(as max of sell in db)

(3) second_name - varchar2(30) not null
Regexp - ^[A-Z]{1}[a-z]{29}+$ - same as first name

(4) university - varchar2(100) not null 
Regexp - ^([A-Z]{1}[a-z]+)+{100}$ - defines arbitrary quantity of groups of words which starts for capital letter

Phones 
-------------
(1) number - varchar2(10) primary key not null
Regexp - ^\+\([0-9]{3}\)\-[0-9\-]{7,12}$ for number like +(097)-134-51-53

(2) country - varchar2(50) not null,
Regexp - ^([A-Z]{1}[a-z]+)+{50}$ - defines arbitrary quantity of groups of words which starts for capital letter

(3) operator - varchar2(20) not null,
Regexp - ^[A-Z]{1}[a-z]{19}+$ - defines first capital letter and 19 small letters(as max of sell in db)

(4) student_id - number(10) not null